Libplanet/KeyStore/Kdfs/Scrypt.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Text.Json;
+using Org.BouncyCastle.Crypto.Digests;
+
+namespace Libplanet.KeyStore.Kdfs
+{
+    /// <summary>
+    /// <a href="https://en.wikipedia.org/wiki/Scrypt">Scrypt</a>.
+    /// </summary>
+    /// <typeparam name="T">PRF (pseudorandom function) to use, e.g.,
+    /// <see cref="Sha256Digest"/>.</typeparam>
+    [SuppressMessage(
+        "Microsoft.StyleCop.CSharp.ReadabilityRules",
+        "SA1402",
+        Justification = "There are just generic & non-generic versions of the same name classes.")]
+    [Pure]
+    public sealed class Scrypt<T> : IKdf
+        where T : GeneralDigest, new()
+    {
+        /// <summary>
+        /// Configures parameters of <a href="https://en.wikipedia.org/wiki/Scrypt">Scrypt</a>.
+        /// </summary>
+        /// <param name="n">CPU/memory cost parameter.</param>
+        /// <param name="salt">A cryptographic salt.</param>
+        /// <param name="keyLength">The desired byte-length of the derived key.
+        /// Corresponds to <c>dkLen</c> except that it's not bit-wise but byte-wise.</param>
+        /// <param name="r">The blocksize parameter.</param>
+        /// <param name="p">Parallelization parameter.</param>
+        public Scrypt(int n, byte[] salt, int keyLength, int r, int p)
+            : this(n, ImmutableArray.Create(salt, 0, salt.Length), keyLength, r, p)
+        {
+        }
+
+        /// <summary>
+        /// Configures parameters of <a href="https://en.wikipedia.org/wiki/Scrypt">Scrypt</a>.
+        /// </summary>
+        /// <param name="n">CPU/memory cost parameter.</param>
+        /// <param name="salt">A cryptographic salt.</param>
+        /// <param name="keyLength">The desired byte-length of the derived key.
+        /// Corresponds to <c>dkLen</c> except that it's not bit-wise but byte-wise.</param>
+        /// <param name="r">The blocksize parameter.</param>
+        /// <param name="p">Parallelization parameter.</param>
+        public Scrypt(int n, in ImmutableArray<byte> salt, int keyLength, int r, int p)
+        {
+            N = n;
+            Salt = salt;
+            KeyLength = keyLength;
+            R = r;
+            P = p;
+        }
+
+        /// <summary>
+        /// CPU/memory cost parameter.
+        /// </summary>
+        public int N { get; }
```

Comment:
How about naming it in more human-readable way, like `Cost`?  As these letters are conventional, it would be good if we mention that in the docs like:

```xml
Corresponds to <c>N</c>.
```